booting_windows/run_vm.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+set -v
+set -euo pipefail
+
+# Get the first two arguments. Default is "none". We need this to avoid
+# "unbound variable" error on $1 or $2
+PARAM1="${1-none}"
+PARAM2="${2-none}"
+
+if [[ "${PARAM1}" == "help" ]]; then
+cat <<USAGETEXT
+  Usage:
+  'run_vm.sh [rebuild_kernel] [rebuild_uroot]'
+     - Run NERF (uroot). Rebuild the kernel or uroot (or both).
+USAGETEXT
+  exit
+fi
+
+MEM=4096 # 4G of DRAM for the virtual machine
+
+pushd "${EFI_WORKSPACE}"
+
+BIOS_PATH=${EFI_WORKSPACE}/downloads/ovmf/usr/share/edk2.git/ovmf-x64/OVMF-pure-efi.fd
+
+if [[ "${PARAM1}" == "rebuild_kernel" ]] || \
+   [[ "${PARAM2}" == "rebuild_kernel" ]]; then
+  pushd linux
+  make -j$(($(nproc)+2))
+  popd
+fi
+
+if [[ "${PARAM1}" == "rebuild_uroot" ]] || \
+   [[ "${PARAM2}" == "rebuild_uroot" ]]; then
+  pushd efi_fs
+  u-root -build=bb -files=bootmgfw.efi
+  popd
+fi
+
+
+# We want to mask all writes to the original Windows image. We therefore
+# create a new snampshot file, which is a delta file. We remove it before
+# every run to make sure we start fresh with the original image.
+WINDOWS_IMAGE="${WORKSPACE}"/win2019.raw
+WINDOWS_DISK=win2019_write_masking.img
+
+if [[ ! -f "${WINDOWS_DISK}" ]]; then
+  qemu-img create -f qcow2 -b "${WINDOWS_IMAGE}" "${WINDOWS_DISK}"
+fi
+
+KERNEL_PATH=linux/arch/x86_64/boot/bzImage
+BOOT_FLAGS="earlyprintk=ttyS0 printk=ttyS0 console=ttyS0 root=/dev/sda1 noefi"
+
+qemu-system-x86_64 -L .                   \
```

Comment:
I had to export DISPLAY=:0 for this to open up its graphical terminal. Failure was correlated with `xterm` also failing in the same way. The error message was:

Unable to init server: Could not connect: Connection refused

Reply:
I added a `export DISPLAY=:0` to run_vm.sh